### PR TITLE
re-implement cloud aware function for e2e test

### PR DESCRIPTION
**Pull request by dem4gus**
### Which issue this PR addresses:
Fixes e2e test failures in USGovCloud environments. [#15119913](https://msazure.visualstudio.com/AzureRedHatOpenShift/_boards/board/t/Chainsaw/Stories/?workitem=15119913)

### What this PR does / why we need it:
Fixes a previously hard-coded string that was causing test failures.

### Test plan for issue:
Run test in USGovCloud environment. Test now re-implements cloud aware logic from controller so confidence is high that it will work correctly.

### Is there any documentation that needs to be updated for this PR?
N/A

### CODE REVIEW

File: test/e2e/operator.go [hunk 1]
```diff
@@ -433,10 +434,20 @@ var _ = Describe("ARO Operator - ImageConfig Reconciler", func() {
...
+		var requiredRegistries []string
+		switch instance.Spec.AZEnvironment {
+		case azureclient.PublicCloud.Environment.Name:
+			requiredRegistries = []string{instance.Spec.ACRDomain, "arosvc." + instance.Spec.Location + ".data." + azure.PublicCloud.ContainerRegistryDNSSuffix}
+
+		case azureclient.USGovernmentCloud.Environment.Name:
+			requiredRegistries = []string{instance.Spec.ACRDomain, "arosvc." + instance.Spec.Location + ".data." + azure.USGovernmentCloud.ContainerRegistryDNSSuffix}
+
+		default:
+			return nil, fmt.Errorf("cloud environment %s is not supported", instance.Spec.AZEnvironment)
```
Comment (cadenmarchese) [inline test/e2e/operator.go:448]: I agree we should fail e2e if environment is unsupported (or if environment is blank/deleted), but not sure if we should fail it during this test specifically.

**Review by cadenmarchese — COMMENTED** [review]
Nice fix, one suggestion - I would personally rather skip the test if environment is blank or unsupported rather than fail the test, but it's an edge case.

File: test/e2e/operator.go (hunk 1 above)
Comment (dem4gus) [reply test/e2e/operator.go:448]: That's a good point but I think I actually like it the way it is now. There's already a check in the `BeforeEach()` that will skip the tests if the feature flag is unset, having an error here will tell us if the flag is enabled on an unsupported cloud environment for whatever reason.

**Review by bennerv — APPROVED** [review]
lgtm

**Review by ross-bryan — APPROVED** [review]
